### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by mhellsten** [SignalR/SignalR#4076]
We're using SignalR 2.2.1 with SignalR.RabbitMq for the backplane, and I noticed the below in the memory dump of  a couple of poorly performing applications. I wanted to confirm that this is abnormal, and ask for hints what could be wrong.

One very interesting detail is that out of three memory dumps at different times, with two applications, the number of the same four types are very close to one million in all three dumps. It doesn't seem like that could be a coincidence.

Thank you much!

```
              MT    Count    TotalSize Class Name
00007ff904d4a4a0      124      8063904 Microsoft.AspNet.SignalR.Messaging.ScaleoutMapping[]
00007ff90312c088   230670     13926648 System.Object[]
00007ff903303980    42023     16887888 System.Collections.Hashtable+bucket[]
00007ff904895b78    82441     23083480 System.Collections.Generic.List`1[[System.Web.HttpApplication+IExecutionStep, System.Web]][]
00007ff903377f08  1006199     24148776 System.WeakReference
00007ff904f4c2d0  1001794     32057408 Microsoft.AspNet.SignalR.Messaging.LocalEventKeyInfo[]
00007ff904f4c240  1001794     40071760 Microsoft.AspNet.SignalR.Messaging.LocalEventKeyInfo
00007ff904d48f60  1001794     40071760 Microsoft.AspNet.SignalR.Messaging.ScaleoutMapping
00007ff9031af9a0    86431     64966081 System.Byte[]
0000002ffd108800   397437     93565742      Free
00007ff9031a4da8  1357711    195108184 System.String
```

**Comment by moozzyk** [comment]
#3031 fixed in 2.2.2 - the default setting was changed from 1000000 tp 65535 and made configurable.

**Comment by anurse** [comment]
This is expected. Messages are not removed from the ring buffer until new messages replace them. As @moozzyk said, this replacement used to happen at 1,000,000 messages, but it has been reduced to 65,535 messages and there is a configuration option for it.
